### PR TITLE
[25] dynamic banners

### PR DESCRIPTION
**Pull request by VitaminTussle**
banners now sync with the ~database~ Json File On The Backend™️, searching is working again (and better), and I added labels to the cabinets so if the banner is nonexistent or weird you can still tell what it is

tied to #25

### CODE REVIEW

File: resources/scripts/gamestatus/index.js [hunk 1]
```diff
@@ -1,80 +1,95 @@
...
+		imgBox.appendChild(grayFadeTag);
+	}
+	imgContainer.appendChild(imgBox);
+	imgContainer.appendChild(document.createElement('br'));
 }
 
 function start() {
 	$.get({
-		url: "https://arkadia-site-api.herokuapp.com/GetAllGameStatuses"
+		url: apiBaseUrl + '/GetAllGameStatuses'
```
Comment (ars75061) [inline resources/scripts/gamestatus/index.js:40]: This is a really good change. have we incorporated the use of the baseURL everywhere else we can?

File: resources/scripts/gamestatus/index.js [hunk 2]
```diff
@@ -1,80 +1,95 @@
...
 	}).done(function(data){
-		cabinets = data;
 		let onlineCabinets = 0;
-		var imgContainer = document.getElementById("imageContainer");
-
+		data = data.sort((a, b) => a.id > b.id ? 1 : -1);
```
Comment (ars75061) [inline resources/scripts/gamestatus/index.js:43]: talk me through what this is doing

File: resources/scripts/gamestatus/index.js [hunk 3]
```diff
@@ -1,80 +1,95 @@
...
 		imageContainer.removeChild(imageContainer.firstChild);
 	}
 
-	var filteredCabinets = [];
-	cabinets.forEach(cabinet => {
-		cabinet.searchTerms.forEach(term => {
-			if(term.includes(searchTerm) && !filteredCabinets.includes(cabinet)) filteredCabinets.push(cabinet);
+	var filteredCabinets = cabinets.map(val => val); // deep copy by value rather than reference
+	let params = searchTerm.split(' ');
+	for(let i = 0; i < params.length; i++) params[i] = params[i].replace(/[^A-Za-z0-9]/g, '');
```
Comment (ars75061) [inline resources/scripts/gamestatus/index.js:68]: Is this standardizing the cabinets for searching?

**Review by ars75061 — CHANGES_REQUESTED** [review]
Looks good, got a few questions though

File: resources/scripts/gamestatus/index.js (hunk 1 above)
Comment (VitaminTussle) [reply resources/scripts/gamestatus/index.js:40]: yeah, the site only accesses the API on first load of the cabinet list page. the api is mostly used on the app side of things for the CUD parts of CRUD

File: resources/scripts/gamestatus/index.js (hunk 2 above)
Comment (VitaminTussle) [reply resources/scripts/gamestatus/index.js:43]: it's just an alphabetical sort A-Z by id. the ids are all just the cabinet names with spaces and special characters removed and set to lowercase (e.g. `Race Drivin'` => `racedrivin`)

File: resources/scripts/gamestatus/index.js (hunk 3 above)
Comment (VitaminTussle) [reply resources/scripts/gamestatus/index.js:68]: not the cabinets, the search terms. the cabinets' relevance terms are already lowercase alphanumeric, so we do the same to the search terms. that way, if someone searches `ms pacman` or `Ms. PacMan` they get the same results

File: resources/scripts/gamestatus/index.js (hunk 1 above)
Comment (ars75061) [reply resources/scripts/gamestatus/index.js:40]: perfect

File: resources/scripts/gamestatus/index.js (hunk 2 above)
Comment (ars75061) [reply resources/scripts/gamestatus/index.js:43]: awesome, LGTM

File: resources/scripts/gamestatus/index.js (hunk 3 above)
Comment (VitaminTussle) [reply resources/scripts/gamestatus/index.js:68]: we also cut out all the spaces so even if someone types `Ms.        Pacman` for some reason, it'll get caught as the search terms just being `['ms', 'pacman']`

**Pull request merged**